### DISCUSSION TITLE
Auto-squaring for dual motors on both X and Y?

### DISCUSSION THREAD

**Discussion by stevempotter** [terjeio/grblHAL#295] [Q&A]
My MPCNC is now working. It has two steppers on each of the X and Y axes. I am planning to move soon to using Phil Barrett's grblHAL breakout board and Teensy 4.1.
I am completely new to grbl. I can not find whether it has the ability to move the two motors of an axis (say, X) independently during homing, to correct for any racking error. Marlin currently does this. If grblHAL does have this capability, how do I implement it? I assume I must use all 5 stepper control channels (X1, X2, Y1, Y2, Z), each with their own min endstop switch, as I am currently doing with Marlin and RAMPS 1.4. Are there software switches or settings I would need to edit?

If it does not have this capability, where is the best place to suggest it for a future version?
And how do you all square your machines accurately in X and Y?

**Comment by terjeio** [comment]
>  If grblHAL does have this capability, how do I implement it?

By changing a #define value from 0 to 1 and recompile for those drivers/booards that support it. Luckily the iMXRT1062 (Teensy 4.x) driver has support for Phils board, enabled in the [board file](https://github.com/grblHAL/iMXRT1062/blob/master/grblHAL_Teensy4/src/T41U5XBB_map.h) for now. The #defines used are near the top (around line 40).

>   I assume I must use all 5 stepper control channels (X1, X2, Y1, Y2, Z), each with their own min endstop switch, as I am currently doing with Marlin and RAMPS 1.4. Are there software switches or settings I would need to edit?

Correct, see above.

> If it does not have this capability, where is the best place to suggest it for a future version?

Please post questions and issues at the new [grblHAL repository](https://github.com/grblHAL), this is going to be phased out.
